### ISSUE TITLE
support defaultDarkMode

### ISSUE THREAD

**Issue by xiaoyunwu** [vuepress/vuepress-next#796]
## Feature request
Right now, the default-them use the prefers-color-scheme via const darkStorage = useStorage('vuepress-color-scheme', 'auto'), so builder does not have explicit control on the default darkMode. 

### Description
support defaultMarkMode which should be auto, light or dark. Where auto is defined by perfers-color-scheme.

### Proposed Solution

There should be a configuration defaultDarkMode so that builder can specify in the config.ts.

**Comment by Mister-Hope** [comment]
1, `'auto'` do fit most situations (you are talking about edge case)
2. You should be able to achieve this by overide `templateBuild` option. (you have workaround)

**Comment by xiaoyunwu** [comment]
How do we do that with templateBuild? I did a google and find nothing. Link?

**Comment by Mister-Hope** [comment]
https://v2.vuepress.vuejs.org/reference/theme-api.html#templatebuild

**Comment by xiaoyunwu** [comment]
Sorry, I meant how-to-use-templateBuild-to-specify-default-darkMode. And
the doc on template build does not have information on how to use it even
for other things. How do we know how to use it?

**Comment by Mister-Hope** [comment]
Place the above template somewhere in your project, and add `templateBuild` option in your user config file, and setting its value to be the path of the file above

**Comment by Mister-Hope** [comment]
>the doc on template build does not have information on how to use it even
> for other things. How do we know how to use it?

Unless the docs is totally missing description or there are missing apis, this kind of problems should be regarded on your side.

We would like to give some simple hints on how you can achieve something, but we do not have reposonability impletement that for you just because you can not understand how to achieve that.

Just as a reminder, this is an open source project, which means we can ( and I will) ignore some personal cutomize help.

**Comment by Mister-Hope** [comment]
Also, if the above way is hacky for you, https://v2.vuepress.vuejs.org/reference/default-theme/extending.html can be another way.

**Comment by xiaoyunwu** [comment]
First of all, I really appreciate your effort in helping around, and make
vuepress great option for creating content site (almost, I did find there
are things missing, like blog2). You are really kind on actions, and I
totally understand you have to maintain a clear boundary.
for  your next suggestion: you mean do something like:
<script setup>
const userMode = localStorage.getItem('vuepress-color-scheme');
      if(!userMode) localStorage.setItem('vuepress-color-scheme', 'xxx');
// xxx is the one you like
const systemDarkMode = window.matchMedia &&
window.matchMedia('(prefers-color-scheme: dark)').matches;
if (userMode === 'dark' || (userMode !== 'light' && systemDarkMode)) {
document.documentElement.classList.toggle('dark', true);
}
import ParentLayout from
***@***.***/theme-default/lib/client/layouts/Layout.vue'
</script>
but I do think both options requires significant increase in term of level
of understanding of how vuepress works (templateBuild is not mentioned here
https://v2.vuepress.vuejs.org/advanced/architecture.html#overview) in the
needed to control default dark mode. But then again, I think it is your
decision in terms of whether defaultdarkmode should be exposed in
accessible fashion like darkmode (should really be darkModeToggle if you
ask me) or not.
Again, thanks for your effort, really prompt, and understood covid is
hurting all of us. Hang in there.
Sean.

**Comment by Mister-Hope** [comment]
You may overide the darkmode toggle button using alias or theme extending, and setting a new default value, the understanding above is incorrrect.

For details you may check the hint in my theme here:

- https://vuepress-theme-hope.github.io/v2/cookbook/advanced/replace.html#replace-components
- https://vuepress-theme-hope.github.io/v2/cookbook/advanced/extend.html#theme-extending

**Comment by xiaoyunwu** [comment]
many thanks, will come back to this tmr.

**Comment by bulletmark** [comment]
I agree with the OP @xiaoyunwu here. I am new to vuepress and find it is odd that there is no config option to default the theme to dark. The templateBuild approach shown here is way too complicated and cumbersome.
